pkg/plugins/ifaces.go

```diff
@@ -148,3 +148,12 @@ type FeatureToggles interface {
 type SignatureCalculator interface {
 	Calculate(ctx context.Context, src PluginSource, plugin FoundPlugin) (Signature, error)
 }
+
+type KeyStore interface {
```

Comment:
Question/nit. Do we see any need of storing additional data besides the key in the store? A key being revoked was discussed in the comments, but maybe a missing key now implicitely means that? In that case or future needs (premature optimizations), might make sense to store a JSON blob instead of a string and use/return struct for updated/retrieving keys. With storing the string now would be hard to change later. Any thoughts?